### PR TITLE
MONGOID-4544 Define #touch on all documents

### PR DESCRIPTION
**Pull request by saghm**

### CODE REVIEW

File: lib/mongoid/touchable.rb [hunk 1]
```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
```
Comment (saghm) [inline lib/mongoid/touchable.rb]: This file is identical to the old `lib/mongoid/association/touchable.rb` (sans the `Association` module)

Comment (p-mongo) [reply lib/mongoid/touchable.rb]: Then why do we have these two identical files in the tree?

File: spec/mongoid/touchable_spec.rb [hunk 2]
```diff
@@ -2,14 +2,31 @@
 
 require "spec_helper"
 
-describe Mongoid::Association::Touchable do
+describe Mongoid::Touchable do
 
   describe "#touch" do
 
+    context "when the document has no relations" do
```
Comment (p-mongo) [inline spec/mongoid/touchable_spec.rb]: This test invokes `touch` on the document. Whether the document has any *associations* (not relations) doesn't matter.

Comment (saghm) [reply spec/mongoid/touchable_spec.rb]: I disagree that it doesn't matter; the entire purpose of this ticket is to add the ability to call `touch` on documents without associations, and that's what this is testing.

**Pull request merged**